tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -312,6 +404,26 @@ protected:
          InvokeReadCallbacks(to);
    }
 
+   /// General implementation of bulk read. Loop over the required range and read values that are required
+   /// and not already present. Derived classes may implement more optimized versions of this method.
+   /// See ReadBulk() for the return value.
+   virtual std::size_t ReadBulkImpl(const RBulkSpec &bulkSpec);
+
+   /// Returns the number of newly available values, that is the number of bools in bulkSpec.fMaskAvail that
+   /// flipped from false to true. As a special return value, kAllSet can be used if all values are read
+   /// independent from the masks.
+   std::size_t ReadBulk(const RBulkSpec &bulkSpec)
+   {
+      if (fIsSimple) {
+         /// For simple types, ignore the mask and memcopy the values into the destination
+         fPrincipalColumn->ReadV(bulkSpec.fFirstIndex, bulkSpec.fCount, bulkSpec.fValues);
+         memset(bulkSpec.fMaskAvail, 1, bulkSpec.fCount);
+         return RBulkSpec::kAllSet;
```

Comment:
what's the advantage of returning the special value here instead of `bulkSpec.fCount` (which would make for a more consistent API and remove the need for callers to check for the special value)?